src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -184,12 +185,16 @@ private DetailAST getEventAst(TreeWalkerAuditEvent event) {
         DetailAST curNode = ast;
         while (curNode != null) {
             DetailAST toVisit = curNode.getFirstChild();
-            if (toVisit != null) {
-                childAstLineNoSet.add(toVisit.getLineNo());
-            }
+            addChildAstLineNo(childAstLineNoSet, toVisit);
             while (curNode != null && toVisit == null) {
                 toVisit = curNode.getNextSibling();
                 curNode = curNode.getParent();
+                if (curNode != null && curNode.equals(ast.getParent())) {
+                    break;
+                }
+            }
+            if (curNode != null && curNode.equals(ast.getParent())) {
+                break;
```

Comment:
same - you can move if-break to main `while` condition. And same with the last if-break in this method.
```java
while (curNode != null && curNode != ast.getParent() && curNode != ast.getNextSibling()) {
```

Reply:
done